src/writer.rs

```diff
@@ -122,13 +122,21 @@ impl FdtWriter {
         };
 
         fdt.align(8);
-        fdt.off_mem_rsvmap = fdt.data.len() as u32;
```

Comment:
Can this `as u32` ever fail? Should we just leave a comment explaining why it can't?

Reply:
This one cannot fail indeed, I think we can just have it as it was before.